### DISCUSSION TITLE
LocalPeopleResults giving 0 results and a HTTP 500 error

### DISCUSSION THREAD

**Discussion by StafPinoy** [microsoft-search/pnp-modern-search#2919] [Q&A]
**Version used**
4.8.0.0

**Describe the bug**
When setting up the "PnP - Search Results" webpart and choosing "LocalPeopleResults" as Result Source Id, I'm getting 0 results. This with all the other settings being the default-value.

When changing the Query Template to "*" I'm getting a HTTP: 500 error on top of this.
![image](https://user-images.githubusercontent.com/60502822/233111714-6b1eeef6-5f86-44fd-9a96-259e57cb4a0c.png)

**Expected behavior**
I would expect to retrieve the users available in the User Profiles in SharePoint.

I'm an SharePoint administrator and I can see that the needed permissions are consented.
![image](https://user-images.githubusercontent.com/60502822/233112208-773e2b3c-e6fc-4313-97d1-53d4202103f4.png)

**Desktop (please complete the following information):**
 - I'm using Edge
 
Thanks in advance for helping

**Comment by wobba** [comment]
Permissions is for the Graph provider, not SharePoint provider. That said, it's excepted to get 0 results when querying for people as a guest. As that would leak tenant private data.

For the 500 error, try to turn of query rules if that is enabled in the web part.

**Reply by StafPinoy** [reply]
Hi @wobba,

I'm using an account from inside the organisation, so no guest account. I meant that I was using an incognito-browser, I have edit the question.

Query rules are disabled already:
![image](https://user-images.githubusercontent.com/60502822/233292944-4151e627-88a4-4db3-9df1-40087cf5c9ec.png)

Staf

**Comment by kasperbolarsen** [comment]
Have you selected the people layout in page 2?

**Reply by StafPinoy** [reply]
Hi Kasper,

Yes, the People Layout is selected.
![image](https://user-images.githubusercontent.com/60502822/233293178-fc989682-b28f-4464-a18a-49e5d87bdd09.png)

**Comment by kasperbolarsen** [comment]
Are you familiar with either the SP editor or the SharePoint search query tool ? If so please verify that the API for people search works.

**Reply by StafPinoy** [reply]
Hi,

I've it tested the SharePoint search query tool (I did not know the existance of it, but it looks very handy).
I also got the issue with the tool, so the issue is not related to the webparts.
![image](https://user-images.githubusercontent.com/60502822/233600278-f03bcd35-a41e-4459-bf58-788a0e4e1421.png)


I will create a Microsoft ticket for this. I will post the outcome in this discussion.

Thanks for helping!

**Comment by kasperbolarsen** [comment]
Indeed, the SP Search tool is an excellent tool, and I have just yesterday released this video about using it when debugging search related issues : youtu.be/SgjweKQ3PeQ
